app/lwip/core/dhcp.c

```diff
@@ -1463,6 +1472,10 @@ dhcp_parse_reply(struct dhcp *dhcp, struct pbuf *p)
         LWIP_ASSERT("len >= decode_len", len >= decode_len);
         decode_idx = DHCP_OPTION_IDX_DNS_SERVER;
         break;
+      case(DHCP_OPTION_NTP):
+        LWIP_ASSERT("len == 4", len == 4);
```

Comment:
What happens if the DHCP server puts two (or more) NTP addresses in that field?

Reply:
Hmm just checked the documentation and saw this option.
Currently I guess it just ignores all but the first. :-(

Reply:
What does the assert do if there is more than one?

Reply:
I guess that the code ought to be
```
        LWIP_ASSERT("len % 4 == 0", len % 4 == 0);
        decode_len = LWIP_MIN(len, 4 * DNS_MAX_SERVERS);
        LWIP_ASSERT("len >= decode_len", len >= decode_len);
```

Reply:
OK found the template in DHCP_OPTION_DNS_SERVER
Will try but my C is not that strong.
Should I add a 
``` C
#define NTP_MAX_SERVERS 2
```
to ``lwipopts.h`` then?`

Or could we just fix NTP_MAX_SERVERS to 1 and be done?

Reply:
@pjsg could you guide the author to finish this PR?

Reply:
Sorry about the delay -- I think that I'm nervous about the whole use of LWIP_ASSERT -- I don't think that it does what anybody expects. I.e. in the production firmware it does nothing. Thus using it to validate (??) data received over wifi is not a good idea. You could argue that we don't need to worry about (in this case) malicious DHCP servers (or other malicious devices that can send DHCP responses) -- but I think that that would be a dangerous argument.

Note that I think that the existing code is dangerous and I'm not sure what the best path forward is.

Reply:
my2cents: We have had those asserts in the code since its inception in 2014. It has seen very few changes since, none of them substantially changed anything and it largely just seems to work. That all doesn't mean you didn't raise a valid point but that I think these concerns shouldn't hold up this PR.
@HHHartmann WDYT?